compiler/transforms/memref_to_snax.py

```diff
@@ -49,6 +44,23 @@ def match_and_rewrite(self, alloc_op: memref.Alloc, rewriter: PatternRewriter):
         total_size_op = None
         ops_to_add = []
 
+        # generate the list of shape ops
+        # either these are constant and must be created,
+        # or they are already present in the memref.alloc
+        # operation arguments
+        shape_ops = []
+        alloc_args = [x.op for x in alloc_op.dynamic_sizes]
+
+        for shape in memref_type.shape.data:
+            if shape.data == -1:
+                # dynamic op
+                shape_ops.append(alloc_args.pop(0))
+            else:
+                # constant op
+                shape_op = Constant.from_int_and_width(shape.data, IndexType())
+                ops_to_add.append(shape_op)
+                shape_ops.append(shape_op)
+
```

Comment:
This is the actually relevant part of this PR
The shape ops are created as a list. For static sizes, a constant op is created. For dynamic sizes, the argument of the `memref.alloc` is used to determine the shape size